tasks/vmaas_sync/package_sync.go

```diff
@@ -218,15 +233,66 @@ func storePackageDetailsFrmPkgListItems(tx *gorm.DB, nameIDs map[string]int, pkg
 	}
 	utils.Log("packages", len(toStore)).Info("Collected packages to store")
 
+	err := storeOrUpdate(tx, toStore)
+	return err
+}
+
+//nolint:funlen
+func storeOrUpdate(tx *gorm.DB, pkgs models.PackageSlice) error {
+	var toUpdate models.PackageSlice
+
+	nameIDEVRAs := make([][]interface{}, 0, len(pkgs))
+	toStore := make(models.PackageSlice, 0, len(pkgs))
+	updateIDs := make(map[int]map[string]int)
+	for _, pkg := range pkgs {
+		nameIDEVRAs = append(nameIDEVRAs, []interface{}{pkg.NameID, pkg.EVRA})
+	}
+
+	if err := tx.Where("(name_id, evra) IN ?", nameIDEVRAs).Find(&toUpdate).Error; err != nil {
+		utils.Log("err", err).Warn("couldn't find packages for update")
+	}
+	for _, u := range toUpdate {
+		if _, has := updateIDs[u.NameID]; !has {
+			updateIDs[u.NameID] = map[string]int{}
+		}
+		updateIDs[u.NameID][u.EVRA] = u.ID
+	}
+
+	// set toUpdate and toStore
+	toUpdate = make(models.PackageSlice, 0, len(pkgs))
+	for _, p := range pkgs {
+		if id, has := updateIDs[p.NameID][p.EVRA]; has {
+			p.ID = id
+			toUpdate = append(toUpdate, p)
+		} else {
+			toStore = append(toStore, p)
+		}
+	}
+
+	// update packages
+	var updErr error
+	for _, u := range toUpdate {
+		updErr = tx.Table("package").Select("description_hash", "summary_hash", "advisory_id").Updates(u).Error
+	}
+	if updErr != nil {
+		storePackagesCnt.WithLabelValues("error").Add(float64(len(toUpdate)))
+		updErr = errors.Wrap(updErr, "Packages update failed")
+	}
+
+	// insert packages
 	tx = database.OnConflictUpdateMulti(tx, []string{"name_id", "evra"},
 		"description_hash", "summary_hash", "advisory_id")
 	if err := tx.CreateInBatches(toStore, chunkSize).Error; err != nil {
 		storePackagesCnt.WithLabelValues("error").Add(float64(len(toStore)))
 		return errors.Wrap(err, "Packages bulk insert failed")
 	}
-	storePackagesCnt.WithLabelValues("success").Add(float64(len(toStore)))
+	if updErr != nil {
+		storePackagesCnt.WithLabelValues("success").Add(float64(len(toStore)))
```

Comment:
Fix this one please and let's merge if Michael does not have anything to add to not waste time with blocked deployments
```suggestion
		storePackagesCnt.WithLabelValues("error").Add(float64(len(toStore)))
```